salt/modules/napalm_yang_mod.py

```diff
@@ -69,7 +69,7 @@ def _get_root_object(models):
 # -----------------------------------------------------------------------------
 
 
-def diff(candidate, running, models):
+def diff(candidate, running, *models):
```

Comment:
The `models` arg is now an argument tuple. Otherwise if you just pass a single string, like it suggests in the docstring/documentation it will break. `_get_root_object` iterates over models and if its a string then it will just iterate over the characters. 
This new logic sets it to a single-value tuple by default and checks for the previous behaviour where a list is passed @mirceaulinic

Reply:
Thanks @tonybaloney - I agree a tuple is better.
After all this work you did here, I am sorry I need to ask for another little change in the NetYANG state (https://docs.saltstack.com/en/latest/ref/states/all/salt.states.netyang.html) that uses the functions from this execution module. Thanks!